airflow-core/src/airflow/executors/executor_loader.py

```diff
@@ -153,6 +154,35 @@ def block_use_of_multi_team(cls):
         if not team_dev_mode or team_dev_mode != "enabled":
             raise AirflowConfigException("Configuring multiple team based executors is not yet supported!")
 
+    @classmethod
+    def _validate_teams_exist_in_database(cls, team_names: set[str]) -> None:
+        """
+        Validate that all specified team names exist in the database.
+
+        :param team_names: Set of team names to validate
+        :raises AirflowConfigException: If any team names don't exist in the database
+        """
+        if not team_names:
+            return
+
+        # Import Session locally to avoid circular import issues
+        from airflow.settings import Session
+
+        with Session() as session:
+            # Get all team names at once to avoid multiple queries
+            existing_teams = {name for (name,) in session.query(Team.name).all()}
```

Comment:
I expect this (list all teams) to be needed in different places. Should we create a static method in model `Team` to retrieve all teams? Returning a dict id -> name

Reply:
Sure, yeah, that's reasonable!

Reply:
Done!